src/utils/daos_hdlr.c

```diff
@@ -2758,44 +2765,58 @@ fs_copy_hdlr(struct cmd_args_s *ap)
 	 * provided
 	 */
 	int			rc = 0;
-	char			src_str[1028];
-	char			dst_str[1028];
+	char			*src_str = NULL;
+	char			*dst_str = NULL;
+	size_t			src_str_len = 0;
+	size_t			dst_str_len = 0;
 	daos_cont_info_t	src_cont_info = {0};
 	daos_cont_info_t	dst_cont_info = {0};
 	struct duns_attr_t	src_dattr = {0};
 	struct duns_attr_t	dst_dattr = {0};
 	struct file_dfs		src_file_dfs = {0};
 	struct file_dfs		dst_file_dfs = {0};
 	struct fs_copy_args	fa = {0};
-	int			src_str_len = 0;
 	char			*name = NULL;
 	char			*dname = NULL;
-	char			dst_dir[MAX_FILENAME];
-	int			path_length = 0;
+	char			*dst_dir = NULL;
 	mode_t			tmp_mode_dir = S_IRWXU;
 
 	file_set_defaults_dfs(&src_file_dfs);
 	file_set_defaults_dfs(&dst_file_dfs);
-	strcpy(src_str, ap->src);
-	rc = fs_copy_parse_path(&src_file_dfs, src_str, &fa.src_p_uuid,
-				&fa.src_c_uuid);
+
+	src_str_len = strlen(ap->src);
```

Comment:
I assume coverity will also complain about strlen rather than strnlen.  I think it would be better to just do something like 

D_STRNDUP(src_str, ap->src, SOME_MAX_LEN_VAR);

I assume fs_copy_parse_path could also take this maximum value ?

Reply:
I don't think Coverity would complain about strlen in this context. What Coverity was complaining about originally was using `strcpy`, since that could overflow a buffer.
And from a discussion with @ashleypittman it's probably better practice to not impose arbitrary limits on the length.
But please correct me if I'm wrong here.